geonode/base/views.py

```diff
@@ -43,6 +43,72 @@
 from geonode.base.forms import BatchEditForm, OwnerRightsRequestForm
 from geonode.base.forms import CuratedThumbnailForm
 from geonode.notifications_helper import send_notification
+from geonode.base.forms import UserAndGroupPermissionsForm
+from django.contrib.auth import get_user_model
+from geonode.tasks.tasks import set_permissions
+from django.contrib.auth.models import Group
+
+
+def user_and_group_permission(request, model):
+    if not request.user.is_superuser:
+        raise PermissionDenied
+
+    model_mapper = {
+        "profile": get_user_model(),
+        "group": Group
+    }
+
+    model_class = model_mapper[model]
```

Comment:
KeyError here. am i missing something?

Reply:
@petrus7, I'll make changes for proper error handling. The reason why I had not done that before was cause values that index `model_mapper` are not unexpected and are hardcoded for each time `user_and_group_permission` is called.

Reply:
oh so the model variable will always be group or profile? . that's fine then